lib/go-tc/deliveryservices.go

```diff
@@ -179,6 +188,11 @@ type DeliveryServiceNullableV30 struct {
 	ServiceCategory    *string `json:"serviceCategory" db:"service_category"`
 }
 
+type DeliveryServiceV31 struct {
+	DeliveryServiceNullableV30
+	MaxRequestHeaderSize *int `json:"maxRequestHeaderSize" db:"max_request_header_size"`
+}
+
 // Deprecated: Use versioned structures only from now on.
 type DeliveryServiceNullable DeliveryServiceNullableV15
```

Comment:
So I had some thoughts @ocket8888 @rob05c. I think we may want to continue using type aliases like this, but only per major version.

Otherwise, let's say we end up with another 5 minor versions in 3.x. That means we will have 5 different `GetDeliveryServiceV*`, `UpdateDeliveryServiceV*`, etc, all in the same client package. I don't think that's what we want the client to look like, right? As a user, I wouldn't want to have to pick between all those minor version methods. I thought we wanted it to auto-downgrade minor versions, which means all the client functions should be using a type alias per major version, e.g. `type DeliveryServiceV3 DeliveryServiceV31`, which we continually update to alias to new minor versions, e.g. `type DeliveryServiceV3 DeliveryServiceV32`. Does that make sense?

Since we already have `DeliveryServiceNullableV30`, maybe that should be the current alias. Then we don't need to add new client methods for V31.

Reply:
That should work out fine, since the only kinds of changes that should break alias property access are the same kinds that would necessitate a major version bump.

Reply:
FWIW, I ended up making an alias in `lib/go-atscfg` for both the latest DS and Server. Just for stability, because it kept changing, and requiring updating hundreds of lines of code. Where using the alias lets me just update that single location, and the few places that broke.

I don't currently have a strong opinion on this. But I think we should do whatever is easy for users to work with. From my recent experience, that seems to be an alias.

> That should work out fine, since the only kinds of changes that should break alias property access are the same kinds that would necessitate a major version bump.

Right, that's the big issue: how does an updating alias work with Go Modules and the library Version Promise?
It's a little bit nebulous what constitutes "breaking." The updating minor alias will only break if people use the nested initialization, e.g. `sv := tc.Server{tc.ServerV3{tc.ServerV2{Foo: foo}, Bar: bar}}` vs `sv := tc.Server{}; sv.Foo = foo; sv.Bar = bar`.

I think I'd be ok with declaring that unsupported in terms of the Go Modules and the library compatibility promise, and documenting so a symbol comment. "Struct alias initialization is only guaranteed forward-compatible for default-initialization."

Reply:
That sounds good to me.